Tried to build the version in your fork:

```
$ rosdep install --from-paths src -n -r -y
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
rosbag_openni_compression: Cannot locate rosdep definition for [openni_wrapper]
libav_image_transport: Cannot locate rosdep definition for [catkinized_libav]
Continuing to install resolvable dependencies...
#All required rosdeps installed successfully
```

and

```
$ catkin_make install
Base path: /home/cdondrup/ros-ws/tmp
Source space: /home/cdondrup/ros-ws/tmp/src
Build space: /home/cdondrup/ros-ws/tmp/build
Devel space: /home/cdondrup/ros-ws/tmp/devel
Install space: /home/cdondrup/ros-ws/tmp/install
####
#### Running command: "cmake /home/cdondrup/ros-ws/tmp/src -DCATKIN_DEVEL_PREFIX=/home/cdondrup/ros-ws/tmp/devel -DCMAKE_INSTALL_PREFIX=/home/cdondrup/ros-ws/tmp/install" in "/home/cdondrup/ros-ws/tmp/build"
####
-- The C compiler identification is GNU 4.6.3
-- The CXX compiler identification is GNU 4.6.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Using CATKIN_DEVEL_PREFIX: /home/cdondrup/ros-ws/tmp/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/hydro
-- This workspace overlays: /opt/ros/hydro
-- Found PythonInterp: /usr/bin/python (found version "2.7.3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python
-- Python version: 2.7
-- Using Debian Python package layout
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/cdondrup/ros-ws/tmp/build/test_results
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found gtest sources under '/usr/src/gtest': gtests will be built
-- catkin 0.5.89
-- BUILD_SHARED_LIBS is on
WARNING: Package "libg2o" does not follow the version conventions. It should not contain leading zeros (unless the number is 0).
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- ~~  traversing 3 packages in topological order:
-- ~~  - catkinized_libav
-- ~~  - rosbag_openni_compression
-- ~~  - libav_image_transport
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- +++ processing catkin package: 'catkinized_libav'
-- ==> add_subdirectory(data_compression/catkinized_libav)
-- +++ processing catkin package: 'rosbag_openni_compression'
-- ==> add_subdirectory(data_compression/rosbag_openni_compression)
-- +++ processing catkin package: 'libav_image_transport'
-- ==> add_subdirectory(data_compression/libav_image_transport)
-- Using these message generators: gencpp;genlisp;genpy
-- Boost version: 1.46.1
-- Found the following Boost libraries:
--   system
--   date_time
--   thread
-- libav_image_transport: 1 messages, 0 services
-- Checking to see if CXX compiler accepts flag -Wl,--version-script,"/home/cdondrup/ros-ws/tmp/build/data_compression/libav_image_transport/class_loader_hide_library_symbols__libav_image_transport.script"
-- Checking to see if CXX compiler accepts flag -Wl,--version-script,"/home/cdondrup/ros-ws/tmp/build/data_compression/libav_image_transport/class_loader_hide_library_symbols__libav_image_transport.script" - yes
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cdondrup/ros-ws/tmp/build
####
#### Running command: "make install -j8 -l8" in "/home/cdondrup/ros-ws/tmp/build"
####
Scanning dependencies of target libav_trunk
Scanning dependencies of target libav_image_transport_gencfg
Scanning dependencies of target std_msgs_generate_messages_py
Scanning dependencies of target std_msgs_generate_messages_lisp
Scanning dependencies of target std_msgs_generate_messages_cpp
[  4%] [  4%] [  4%] [  8%] Built target std_msgs_generate_messages_lisp
Creating directories for 'libav_trunk'
[ 13%] [ 13%] Generating dynamic reconfigure files from cfg/libavPublisher.cfg: /home/cdondrup/ros-ws/tmp/devel/include/libav_image_transport/libavPublisherConfig.h /home/cdondrup/ros-ws/tmp/devel/lib/python2.7/dist-packages/libav_image_transport/cfg/libavPublisherConfig.py
Generating dynamic reconfigure files from cfg/libavSubscriber.cfg: /home/cdondrup/ros-ws/tmp/devel/include/libav_image_transport/libavSubscriberConfig.h /home/cdondrup/ros-ws/tmp/devel/lib/python2.7/dist-packages/libav_image_transport/cfg/libavSubscriberConfig.py
Built target std_msgs_generate_messages_py
Built target std_msgs_generate_messages_cpp
Scanning dependencies of target libav_image_transport_generate_messages_py
Scanning dependencies of target libav_image_transport_generate_messages_cpp
Scanning dependencies of target libav_image_transport_generate_messages_lisp
[ 17%] [ 21%] Generating Lisp code from libav_image_transport/Packet.msg
Generating Python from MSG libav_image_transport/Packet
[ 26%] [ 30%] Generating C++ code from libav_image_transport/Packet.msg
No download step for 'libav_trunk'
Generating reconfiguration files for libavPublisher in libav_image_transport
Generating reconfiguration files for libavSubscriber in libav_image_transport
Wrote header file in /home/cdondrup/ros-ws/tmp/devel/include/libav_image_transport/libavPublisherConfig.h
Wrote header file in /home/cdondrup/ros-ws/tmp/devel/include/libav_image_transport/libavSubscriberConfig.h
[ 34%] [ 39%] No patch step for 'libav_trunk'
No update step for 'libav_trunk'
[ 39%] Built target libav_image_transport_gencfg
[ 43%] [ 47%] Performing configure step for 'libav_trunk'
/bin/sh: 1: /home/cdondrup/ros-ws/tmp/src/data_compression/catkinized_libav/libav_trunk/configure: not found
make[2]: *** [data_compression/catkinized_libav/libav_trunk-prefix/src/libav_trunk-stamp/libav_trunk-configure] Error 127
make[1]: *** [data_compression/catkinized_libav/CMakeFiles/libav_trunk.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 47%] Generating Python msg __init__.py for libav_image_transport
Built target libav_image_transport_generate_messages_lisp
[ 47%] Built target libav_image_transport_generate_messages_py
[ 47%] Built target libav_image_transport_generate_messages_cpp
make: *** [all] Error 2
Invoking "make" failed
```

Am I missing anything?

--- a/README.md
+++ b/README.md
@@ -4,3 +4,12 @@ data_compression
 Video encoding for OpenNI topics to record them to a rosbag and replay through reconstruction of point clouds.
 
 Includes libav_image_transport by Dominique Hunziker: https://github.com/rapyuta/libav_image_transport.
+
+## Compiling
+
+Before doing `catkin_make` on the containing workspace, you need to fetch the submodules one time
+
+* `git submodule init`
+* `git submodule update`
+
+within the git repository.

--- a/catkinized_libav/CMakeLists.txt
+++ b/catkinized_libav/CMakeLists.txt
@@ -19,7 +19,7 @@ LIBRARIES libav_wrap
 ExternalProject_Add(libav_trunk
   DOWNLOAD_COMMAND ""
   SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/libav_trunk"
-  CONFIGURE_COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/libav_trunk/configure" --enable-shared --enable-libx264 --enable-gpl --prefix="${CATKIN_DEVEL_PREFIX}"
+  CONFIGURE_COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/libav_trunk/configure" --enable-shared --enable-gpl --prefix="${CATKIN_DEVEL_PREFIX}"#--enable-libx264 
 )
 
 add_library(libav_wrap src/wrap_lib.cc)

--- a/catkinized_libav/package.xml
+++ b/catkinized_libav/package.xml
@@ -30,7 +30,7 @@
   <!-- Examples: -->
   <buildtool_depend>catkin</buildtool_depend>
   <build_depend>yasm</build_depend>
-  <build_depend>libx264-dev</build_depend>
+  <!-- <build_depend>libx264-dev</build_depend> -->
 
   <!-- The export tag contains other, unspecified, tags -->
   <export>

